### PR TITLE
[MTM-44351]: Migrate java agent to Java 11. Allow to build examples w…

### PR DESCRIPTION
**Pull request by c8y-kh4mell**
…ith Java 17

### CODE REVIEW

File: snmp/snmp-device-gateway/src/main/java/com/cumulocity/agent/snmp/persistence/AbstractQueue.java [hunk 1]
```diff
@@ -74,13 +81,6 @@ public AbstractQueue(String queueName, File persistenceFolder) {
 
         log.info("Creating/Loading '{}' Queue, backed by the folder '{}'", this.name, persistenceFolder.getPath());
 
-        // Since the rolling cycle is set to minutely and we also delete the old files periodically,
-        // we are fine with not shrinking the data files.
-        // Hence we disable the data file shrinking, unless it is explicitly requested by setting the
-        // system property "chronicle.queue.disableFileShrinking"
-        QueueFileShrinkManager.DISABLE_QUEUE_FILE_SHRINKING = Boolean.parseBoolean(
-                System.getProperty("chronicle.queue.disableFileShrinking", Boolean.TRUE.toString()));
-
```
Comment (bbyreddy) [inline snmp/snmp-device-gateway/src/main/java/com/cumulocity/agent/snmp/persistence/AbstractQueue.java:83]: - [ ] Any particular reason why this code is removed? The static code added to the class is not going to substitute the setting of the QueueFileShrinkManager.DISABLE_QUEUE_FILE_SHRINKING flag.

File: snmp/pom.xml [hunk 2]
```diff
@@ -155,13 +155,10 @@
...
-				<version>${chronicle.queue.version}</version>
-			</dependency>
-			<dependency>
-				<groupId>net.openhft</groupId>
-				<artifactId>chronicle-map</artifactId>
-				<version>${chronicle.map.version}</version>
+				<artifactId>chronicle-bom</artifactId>
+				<type>pom</type>
+				<scope>import</scope>
+				<version>2.22.132</version>
```
Comment (bbyreddy) [inline snmp/pom.xml:159]: - [x] With the introduction of these changes and dependency on chronicle-bom, I guess we don't need the two constants chronicle.map.version and chronicle.queue.version. Can these be removed as they are still pointing to old versions and is also misleading?
- [x] Also with this change, aren't we pulling and bundling many more chronicle libraries which are not used by the agent? Earlier SNMP agent depended on only two chronicle libraries chronicle-queue and chronicle-map.

Comment (c8y-kh4mell) [reply snmp/pom.xml:159]: BOMs in general when inserted into dependencyManagement section are used to manage dependencies versions. It does not lead to dependency import, it only makes sure a valid versions (compatible with each other) are used in application. To summarize: no more dependencies are imported than previously.

File: snmp/snmp-device-gateway/src/main/java/com/cumulocity/agent/snmp/persistence/AbstractQueue.java [hunk 3]
```diff
@@ -74,13 +81,6 @@ public AbstractQueue(String queueName, File persistenceFolder) {
 
         log.info("Creating/Loading '{}' Queue, backed by the folder '{}'", this.name, persistenceFolder.getPath());
 
-        // Since the rolling cycle is set to minutely and we also delete the old files periodically,
-        // we are fine with not shrinking the data files.
-        // Hence we disable the data file shrinking, unless it is explicitly requested by setting the
-        // system property "chronicle.queue.disableFileShrinking"
-        QueueFileShrinkManager.DISABLE_QUEUE_FILE_SHRINKING = Boolean.parseBoolean(
```
Comment (c8y-kh4mell) [inline snmp/snmp-device-gateway/src/main/java/com/cumulocity/agent/snmp/persistence/AbstractQueue.java:81]: `DISABLE_QUEUE_FILE_SHRINKING` is now final and not possible to assign value to it

File: snmp/pom.xml [hunk 4]
```diff
@@ -46,8 +46,6 @@
 		<cometd.version>3.0.10</cometd.version>
 		<svenson.version>1.5.8</svenson.version>
 		<jetty.version>9.4.43.v20210629</jetty.version>
-		<chronicle.map.version>3.17.6</chronicle.map.version>
```
Comment (c8y-kh4mell) [inline snmp/pom.xml:49]: @bbyreddy removed

File: snmp/pom.xml (hunk 2 above)
Comment (bbyreddy) [reply snmp/pom.xml:159]: Got it, thanks Kamil.

**Review by bbyreddy — DISMISSED** [review]
Approved from SNMP Agent side changes.

**Review by AleksanderFiszCumulocity — DISMISSED** [review]

**Review by jkot-c8y — DISMISSED** [review]

**Review by karol-c8y — DISMISSED** [review]

**Review by karol-c8y — APPROVED** [review]

**Review by jkot-c8y — APPROVED** [review]

**Pull request merged**